File: Backend/authservice/src/main/java/com/example/demo/entities/Task.java
package com.example.demo.entities;

import java.time.LocalDate;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.*;

@Entity
@Table(name = "task")
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int taskId;

    @Column(nullable = false)
    private String tname;

    @Column(length = 500)
    private String tdescription;

    @ManyToOne
    @JoinColumn(name = "pid")
    @JsonIgnoreProperties("tasks")
    private Project project;

    @Column(nullable = false)
    private String status;

    @ManyToOne
    @JoinColumn(name = "uid")
    @JsonIgnoreProperties("tasks")
    private User user;

    private LocalDate start_date;
    private LocalDate end_date;

    public Task() {
        super();
    }

    // getters and setters

    public int getTask_id() {
        return taskId;
    }

    public void setTask_id(int task_id) {
        this.taskId = task_id;
    }

    public String getTname() {
        return tname;
    }

    public void setTname(String tname) {
        this.tname = tname;
    }

    public String getTdescription() {
        return tdescription;
    }

    public void setTdescription(String tdescription) {
        this.tdescription = tdescription;
    }

    public Project getProject() {
        return project;
    }

    public void setProject(Project project) {
        this.project = project;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public LocalDate getStart_date() {
        return start_date;
    }

    public void setStart_date(LocalDate start_date) {
        this.start_date = start_date;
    }

    public LocalDate getEnd_date() {
        return end_date;
    }

    public void setEnd_date(LocalDate end_date) {
        this.end_date = end_date;
    }
}

File: Backend/employee_service/employee_service/Models/Task.cs
using System;
using System.Collections.Generic;

namespace employee_service.Models;

public partial class Task
{
    public long TaskId { get; set; }

    public string Tname { get; set; } = null!;

    public string Tdescription { get; set; } = null!;

    public int Pid { get; set; }

    public string Status { get; set; } = null!;

    public int Uid { get; set; }

    public DateOnly StartDate { get; set; }

    public DateOnly EndDate { get; set; }

    public int? DomainId { get; set; }

    public string? Priority { get; set; }

    public virtual Domain? Domain { get; set; }

    public virtual Project PidNavigation { get; set; } = null!;

    public virtual User UidNavigation { get; set; } = null!;
}
